### ISSUE TITLE
Add a timeout for SASL authentication

### ISSUE THREAD

**Issue by nyuszika7h** [ProgVal/Limnoria#854]
For example, if services are offline and the bot tries to authenticate with SASL, the IRCd won't reply to the `AUTHENTICATE` command, so the bot can't connect. A (configurable) timeout which most IRC clients have would solve this issue.

**Comment by Mikaela** [comment]
This issue prevents connecting to OFTC if certfp is enabled and OFTC only supports certfp and `/ns` or `/nickserv`.

**Comment by Mikaela** [comment]
Actually not sure on the previous comment. Attached paste: https://paste.openttdcoop.org/pbsds9wq6#line-22

```
DEBUG 2015-01-18T17:13:10 supybot Outgoing message (oftc): NICK :Soapybot
DEBUG 2015-01-18T17:13:10 supybot Irc.takeMsg throttling.
DEBUG 2015-01-18T17:13:10 supybot Incoming message (oftc): :solenoid.oftc.net NOTICE AUTH :*** Looking up your hostname...
DEBUG 2015-01-18T17:13:10 supybot Incoming message (oftc): :solenoid.oftc.net NOTICE AUTH :*** Checking Ident
DEBUG 2015-01-18T17:13:10 supybot Irc.takeMsg throttling.
DEBUG 2015-01-18T17:13:10 supybot Incoming message (oftc): :solenoid.oftc.net NOTICE AUTH :*** Found your hostname
DEBUG 2015-01-18T17:13:10 supybot Irc.takeMsg throttling.
DEBUG 2015-01-18T17:13:10 supybot Irc.takeMsg throttling.
DEBUG 2015-01-18T17:13:10 supybot Irc.takeMsg throttling.
DEBUG 2015-01-18T17:13:10 supybot Irc.takeMsg throttling.
DEBUG 2015-01-18T17:13:10 supybot Incoming message (oftc): :solenoid.oftc.net NOTICE AUTH :*** No Ident response
DEBUG 2015-01-18T17:13:10 supybot Irc.takeMsg throttling.
DEBUG 2015-01-18T17:13:11 supybot Outgoing message (oftc): CAP REQ :sasl
DEBUG 2015-01-18T17:13:11 supybot Irc.takeMsg throttling.
DEBUG 2015-01-18T17:13:11 supybot Irc.takeMsg throttling.
DEBUG 2015-01-18T17:13:11 supybot Irc.takeMsg throttling.
DEBUG 2015-01-18T17:13:11 supybot Irc.takeMsg throttling.
DEBUG 2015-01-18T17:13:11 supybot Incoming message (oftc): :solenoid.oftc.net CAP Soapybot NAK :sasl
WARNING 2015-01-18T17:13:11 supybot oftc: Server refused capability 'sasl'
DEBUG 2015-01-18T17:13:11 supybot Irc.takeMsg throttling.
DEBUG 2015-01-18T17:13:12 supybot Outgoing message (oftc): CAP REQ :account-notify
DEBUG 2015-01-18T17:13:12 supybot Irc.takeMsg throttling.
DEBUG 2015-01-18T17:13:12 supybot Irc.takeMsg throttling.
DEBUG 2015-01-18T17:13:12 supybot Irc.takeMsg throttling.
DEBUG 2015-01-18T17:13:12 supybot Irc.takeMsg throttling.
DEBUG 2015-01-18T17:13:12 supybot Incoming message (oftc): :solenoid.oftc.net CAP Soapybot NAK :account-notify
WARNING 2015-01-18T17:13:12 supybot oftc: Server refused capability 'account-notify'
DEBUG 2015-01-18T17:13:12 supybot Irc.takeMsg throttling.
DEBUG 2015-01-18T17:13:13 supybot Outgoing message (oftc): CAP REQ :extended-join
DEBUG 2015-01-18T17:13:13 supybot Irc.takeMsg throttling.
DEBUG 2015-01-18T17:13:13 supybot Irc.takeMsg throttling.
DEBUG 2015-01-18T17:13:13 supybot Irc.takeMsg throttling.
DEBUG 2015-01-18T17:13:13 supybot Irc.takeMsg throttling.
DEBUG 2015-01-18T17:13:13 supybot Incoming message (oftc): :solenoid.oftc.net CAP Soapybot NAK :extended-join
WARNING 2015-01-18T17:13:13 supybot oftc: Server refused capability 'extended-join'
DEBUG 2015-01-18T17:13:13 supybot Irc.takeMsg throttling.
DEBUG 2015-01-18T17:13:14 supybot Outgoing message (oftc): USER soapy 0 * :Soapybot
DEBUG 2015-01-18T17:13:40 supybot Incoming message (oftc): ERROR :Closing Link: nurionis.co.uk (Registration timed out)
WARNING 2015-01-18T17:13:40 supybot Error message from oftc: Closing Link: nurionis.co.uk (Registration timed out)
```

**Comment by nyuszika7h** [comment]
@Mikaela Can you test whether https://github.com/nyuszika7h/limnoria/commit/b4eb72d578d676d17e832ba6f011767f21c6943e fixes your issue, please?

**Comment by Mikaela** [comment]
@nyuszika7h Sadly I cannot as I am going to prepare for bed and my bot isn't at OFTC. This paste came from someone on \#supybot at OFTC, you can probably try to contact them about it.

**Comment by GLolol** [comment]
This should be fixed by #1013.
